main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -707,43 +717,38 @@ public boolean onPrepareOptionsMenu(final Menu menu) {
         final boolean isOffline = type == CacheListType.OFFLINE;
         final boolean isEmpty = cacheList.isEmpty();
         final boolean isConcrete = isConcreteList();
+        final boolean isNonDefaultList = isConcrete && listId != StoredList.STANDARD_LIST_ID;
+        final List<CacheListApp> listNavigationApps = CacheListApps.getActiveApps();
 
         try {
+            // toplevel menu items
+            setEnabled(menu, R.id.menu_show_on_map, !isEmpty);
+            setEnabled(menu, R.id.menu_filter, search != null && search.getCount() > 0);
+            setVisibleEnabled(menu, R.id.menu_sort, !isHistory, !isEmpty);
             if (adapter.isSelectMode()) {
                 menu.findItem(R.id.menu_switch_select_mode).setTitle(res.getString(R.string.caches_select_mode_exit))
                         .setIcon(R.drawable.ic_menu_clear_playlist);
             } else {
                 menu.findItem(R.id.menu_switch_select_mode).setTitle(res.getString(R.string.caches_select_mode))
                         .setIcon(R.drawable.ic_menu_agenda);
             }
-            menu.findItem(R.id.menu_invert_selection).setVisible(adapter.isSelectMode());
-
-            setVisible(menu, R.id.menu_show_on_map, !isEmpty);
-            setVisible(menu, R.id.menu_filter, search != null && search.getCount() > 0);
-            setVisible(menu, R.id.menu_switch_select_mode, !isEmpty);
-            setVisible(menu, R.id.menu_create_list, isOffline);
-
-            setVisible(menu, R.id.menu_sort, !isEmpty && !isHistory);
-            setVisible(menu, R.id.menu_caches, !isEmpty);
-            setVisible(menu, R.id.menu_refresh_stored, !isEmpty);
-            setVisible(menu, R.id.menu_drop_caches, !isEmpty && (isHistory || isOffline));
-            setVisible(menu, R.id.menu_delete_events, isConcrete && !isEmpty && containsPastEvents());
-            setVisible(menu, R.id.menu_move_to_list, isOffline && !isEmpty);
-            setVisible(menu, R.id.menu_copy_to_list, isOffline && !isEmpty);
-            setVisible(menu, R.id.menu_remove_from_history, !isEmpty && isHistory);
-            setVisible(menu, R.id.menu_clear_offline_logs, !isEmpty && (isHistory || isOffline) && containsOfflineLogs());
-            setVisible(menu, R.id.menu_import, isOffline);
-            setVisible(menu, R.id.menu_import_web, isOffline);
-            setVisible(menu, R.id.menu_import_gpx, isOffline);
-            setVisible(menu, R.id.menu_export, !isEmpty && (isHistory || isOffline));
-            setVisible(menu, R.id.menu_make_list_unique, !isEmpty && isOffline && listId != PseudoList.ALL_LIST.id);
+            setEnabled(menu, R.id.menu_switch_select_mode, !isEmpty);
+            setEnabled(menu, R.id.menu_invert_selection, adapter.isSelectMode());
```

Comment:
This one might lead to confusion. A lot of people never use the selection mode at all, but will always see a menu item "Invert selection" now. I would exchange enabled for visible here.